### DISCUSSION TITLE
Theia IDE 1.57 preview testing

### DISCUSSION THREAD

**Discussion by jfaltermeier** [eclipse-theia/theia#14643] [Release Announcements] [answered]
The new version 1.57.0 of the Theia IDE is available on the preview channel now, please join the preview testing! You can download it here: [Linux](https://download.eclipse.org/theia/ide-preview/1.57.0/linux/TheiaIDE.AppImage), [Mac](https://download.eclipse.org/theia/ide-preview/1.57.0/macos/TheiaIDE.dmg), [Windows](https://download.eclipse.org/theia/ide-preview/1.57.0/windows/TheiaIDESetup.exe). You can update your existing installation by setting the preference updates.channel to preview. Please respond here when you were able to test the preview without finding blockers, by commenting with a ✔️. If you find any issues, please mention them in this thread and report them as an issue once confirmed by other testers.

**Comment by jfaltermeier** [comment]
I noticed that the fixes from https://github.com/eclipse-theia/theia/pull/14557 and https://github.com/eclipse-theia/theia/pull/14548 do not seem to work with the Preview.

During the release, the native dependencies were rebuilt and I verified that the updated ones were uploaded on the release (https://github.com/eclipse-theia/theia/actions/runs/12352417683).

This might not be a blocker, though, since it was not working with 1.56 as well.

**Reply by sgraband** [reply]
I was also able to reproduce that #14548 is not fixed in the Theia IDE. Haven't tested the other one yet.

**Comment by sgraband** [comment]
:heavy_check_mark: Apart from the issue mentioned earlier, which should not be a blocker.

**Comment by sdirix** [comment]
There is a blocker: The new ScanOSS functionality does not work in the Preview build

![image](https://github.com/user-attachments/assets/188ca10f-205d-49f8-a198-757a777cffa9)

I investigated the issue a bit. It stems from `node-fetch` [performing some constructor name checks](https://github.com/node-fetch/node-fetch/blob/9b9d45881e5ca68757077726b3c0ecf8fdca1f29/src/request.js#L65). We bundle and minimize the backend, therefore the names are stripped from the code and `node-fetch` fails. I'm working on a pragmatic fix.

**Reply by sdirix** [reply]
I found two fixes which could be applied:

**Option1**: remove class names mangling via terser options, e.g.
```
new TerserPlugin({
        terserOptions: {
          keep_classnames: true,
        },
      })
```

This could be applied to the webpack configuration of the Theia IDE and potentially into the webpack generator.

**Option2**: Increase `node-fetch` by a major version from `2.x` to `3.x`. This has to be applied via `resolutions`. I didn't see a regression when doing this, however `node-fetch` *does* print some deprecation warnings in its usage.

**Reply by sdirix** [reply]
I created:
- https://github.com/eclipse-theia/theia/pull/14650 for Theia
- https://github.com/eclipse-theia/theia-ide/pull/434 for the Theia IDE

using the `terser` customization approach, just a bit more specific:

```ts
terserOptions: {
    keep_classnames: /AbortSignal/
}
```

For the Theia IDE itself it's sufficient to merge the PR for the Theia IDE.

It would not be sufficient to just merge the PR for Theia and consume an updated version in the Theia IDE, as we already override the `TerserPlugin` there. So the Theia IDE PR is needed in any case.

**Reply by sdirix** [reply]
There was one more issue only occuring on Windows. It was fixed here: https://github.com/eclipse-theia/theia/pull/14653

**Reply by sdirix** [reply]
With that, I didn't see any additional blocker :heavy_check_mark:

**Comment by tsmaeder** [comment]
Sorry for the late feedback, I was not using the IDE enough this morning to give good feedback. It's a ✔️  from my side.

**Comment by JonasHelming** [comment]
:white_check_mark: except the SCANOSS blocker

**Comment by rschnekenbu** [comment]
✔️for me. Looking good

**Comment by tsmaeder** [comment]
Update to interested parties: we've promoted the current preview to the stable channel as the blocker only affect new features. We will provide an updated build where the identified issues are fixed later this week.

**Comment by jfaltermeier** [comment]
A new preview (1.57.100) based on Theia 1.57.1 was published: [Linux](https://download.eclipse.org/theia/ide-preview/1.57.100/linux/TheiaIDE.AppImage), [Mac](https://download.eclipse.org/theia/ide-preview/1.57.100/macos/TheiaIDE.dmg), [Windows](https://download.eclipse.org/theia/ide-preview/1.57.100/windows/TheiaIDESetup.exe)

**Reply by sdirix** [reply]
I can confirm that the SCANOSS issues on Windows are gone :heavy_check_mark:

**Reply by JonasHelming** [reply]
Me, too!

**Reply by sgraband** [reply]
LGTM :heavy_check_mark:

**Reply by JonasHelming** [reply]
Same: :heavy_check_mark:

**Reply by tsmaeder** [reply]
+1 from me: ✔️

**Reply by jfaltermeier** [reply]
✔️ Saw no issues after upgrading from ubuntu 22.04 to 24.04 as well

**Comment by jfaltermeier** [comment, marked as answer]
1.57.100 is promoted to the stable channel. Mirrors will catch up in the next hours

**Comment by JonasHelming** [comment]
[News & Noteworthy](https://eclipsesource.com/blogs/2024/12/20/eclipse-theia-1-57-release-news-and-noteworthy/)
